kamu-core/src/infra/repos/object_repository_in_memory.rs

```diff
@@ -56,7 +67,15 @@ impl ObjectRepository for ObjectRepositoryInMemory {
     }
 
     async fn get_stream(&self, _hash: &Multihash) -> Result<Box<AsyncReadObj>, GetError> {
-        unimplemented!()
+        panic!("get_stream not allowed for in-memory repository");
+    }
+
+    async fn get_download_url(
+        &self,
+        _hash: &Multihash,
+        _opts: DownloadOpts,
+    ) -> Result<GetDownloadUrlResult, GetDownloadUrlError> {
+        panic!("get_download_url not allowed for in-memory repository");
```

Comment:
Why not return `Err(GetDownloadUrlError::NotSupported)` instead of panic?

Reply:
In-memory repository is only used by tests and panics in many if the overrides.